File: 035__Search_Insert_Position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        low = 0
        high = len(nums) - 1
        if target > nums[high]:
            return high + 1
        while low < high - 1:
            mid = (low + high) / 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                high = mid
            else:
                low = mid

        if target == nums[low]:
            return low
        elif target > nums[low]:
            return low + 1;
        else:
            return 0

    

File: Java/035__Search_Insert_Position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        if (nums == null || nums.length == 0 || nums[0] == target) return 0;
		if (target == nums[nums.length - 1]) return nums.length - 1;
		if (target > nums[nums.length - 1]) return nums.length;
        int l = 0, r = nums.length;
        while (l <= r) {
            int mid = (l + r) / 2;
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] > target)
                r = mid - 1;
            else
                l = mid + 1;
        }
        return r + 1;
    }
}
